app/scripts/ui.js

```diff
@@ -11,19 +11,62 @@ import Eth from 'ethjs';
 import EthQuery from 'eth-query';
 import StreamProvider from 'web3-stream-provider';
 import log from 'loglevel';
+import fetch from 'node-fetch';
 import launchMetaMaskUi from '../../ui';
 import {
   ENVIRONMENT_TYPE_FULLSCREEN,
   ENVIRONMENT_TYPE_POPUP,
 } from '../../shared/constants/app';
+import { SUPPORT_LINK } from '../../ui/helpers/constants/common';
 import ExtensionPlatform from './platforms/extension';
 import { setupMultiplex } from './lib/stream-utils';
 import { getEnvironmentType } from './lib/util';
 import metaRPCClientFactory from './lib/metaRPCClientFactory';
+import getFirstPreferredLangCode from './lib/get-first-preferred-lang-code';
 
 start().catch(log.error);
 
 async function start() {
+  const preferredLocale = await getFirstPreferredLangCode();
+
+  const url = browser.runtime.getURL(
+    `../_locales/${preferredLocale}/messages.json`,
+  );
+  const messages = await (await fetch(url)).json();
+
+  function displayCriticalError(container, err, msg) {
+    const html = `
+    <style>
+      #app-content {margin: 10px;}
+      .critical-error {
+        color:red; 
+        background-color:yellow; 
+        text-align: left
+      }
+      blockquote {
+        margin: 20px 0 20px 5px; 
+        padding-left: 5px; 
+        border-left: 1px solid grey
+      }  
+      a {color: blue}
+      a:hover {color: green}
+    </style>
+    <div class="critical-error">
```

Comment:
We can add these elements directly to `home.html`. I don't think there's any need to set `innerHTML` at runtime except to set a localized message.